doc/adr/0001-reactive-reassignment.md

```diff
@@ -0,0 +1,49 @@
+# 1. Reactive reassignment
+
+Date: 2018-11-21
+
+## Status
+
+Accepted
+
+## Context
+
+In current architecture, we receive always a batch of products. For this batch, we check if any products need reassignment.
+In order to find out, we search for conflicting products with query by SKUs from variants and slugs from all locale and both current and staged.
+This query can get too big for GET request and cannot be used.  
+
+## Decision
+
+Variant reassignment will be called only in case of errors and it will solve only the failed product.
+
+## Consequences
+Refactoring needs to be done both in reassignment module and also modules that uses reassignment (right now only sphere-product-import).
+
+#### sphere-product-import changes
+1. Catch errors and on errors with 400 HTTP status call variant reassignment with the failed product. Reassignment will run in these cases:
```

Comment:
`400 HTTP status call` - but also 409 (concurrent modification)

Reply:
Product import has a repeater which recalculates and resends update actions on 409 error so this should be handled already